codersdk/aitasks.go

```diff
@@ -102,6 +102,21 @@ const (
 	TaskStateFailed   TaskState = "failed"
 )
 
+func TaskStateFromWorkspaceAppStatus(was WorkspaceAppStatusState) TaskState {
+	switch was {
+	case WorkspaceAppStatusStateWorking:
+		return TaskStateWorking
+	case WorkspaceAppStatusStateIdle:
+		return TaskStateIdle
+	case WorkspaceAppStatusStateComplete:
+		return TaskStateComplete
+	case WorkspaceAppStatusStateFailure:
+		return TaskStateFailed
+	default:
+		return TaskState("unknown")
```

Comment:
If this is needed, we should add the type rather than return an unexpected state.

```suggestion
		return TaskStateUnknown
```